routes/post.js

```diff
@@ -12,86 +12,122 @@ function resError(res, message){
+		}
 
-						// Try to send emails to participants.
-						var participantIds = convo.participants.map(function (user){
-							return user._id;
-						});
-						var promise = User.find({ _id: { $in: participantIds }}).exec();
-						promise.then(function (participants){
-							var currentUser = participants.filter(function (user){
-								return user._id.toString() === req.body.userId;
-							})[0];
-							var emails = participants.map(function (user){return user.username});
-							var subject = 'GC: ' + currentUser.firstName + ' responded to the question: ' + convo.question;
-							if (subject.length > 70){
-								subject = subject.slice(0, 70) + '...';
-							}
-							var contentSnippet = content.slice(0, 140) + '...';
+		// Add the post to the conversation's discussion and save
+	    Conversation.findOne( { _id: conversationId }, function (err, convo, count){
+	    	if (err || !convo){
+	    		return onFailure("Could not find messages collection in DB.");
+			}
 
-							Mailman.sendMail({
-								recipients: emails,
-								subject: subject,
-								html: 'Hello,<br><br>In a follow-up to the conversation<br><h3>' +
-									convo.question + '</h3>' + currentUser.firstName +
-									' added: ' + contentSnippet + '<br>' +
-									'<br>Continue the conversation on <a href="http://goldenconversations.heroku.com/">Golden Conversations</a>.',
-								callback: function(){
-									console.log('New conversation notification emails have been sent.');
-								}
-							});
-						});
+			convo.discussion.push(post._id);
+	    	convo.save(function(err, message){
+	    		if (err){
+	    			onFailure("Could not save message into db.");
+				} else {
+					onSuccess();
+				}
+			});
+
+			// Try to send emails to participants.
+			var participantIds = convo.participants.map(function (user){
+				return user._id;
+			});
+			var promise = User.find({ _id: { $in: participantIds }}).exec();
+			promise.then(function (participants){
+				var currentUser = participants.filter(function (user){
+					return user._id.toString() === userId;
+				})[0];
+				var emails = participants.map(function (user){return user.username});
+				var subject = 'GC: ' + currentUser.firstName + ' responded to the question: ' + convo.question;
+				if (subject.length > 70){
+					subject = subject.slice(0, 70) + '...';
+				}
+				var contentSnippet = content.slice(0, 140) + '...';
+
+				Mailman.sendMail({
+					recipients: emails,
+					subject: subject,
+					html: 'Hello,<br><br>In a follow-up to the conversation<br><h3>' +
+						convo.question + '</h3>' + currentUser.firstName +
+						' added: ' + contentSnippet + '<br>' +
+						'<br>Continue the conversation on <a href="http://goldenconversations.heroku.com/">Golden Conversations</a>.',
+					callback: function(){
+						console.log('New conversation notification emails have been sent.');
 					}
-			    });
-			}
-		});
-	}
-};
+				});
+			});
+		
```

Comment:
If I'm not mistaken, we might need a closing parenthesis here. It seems like there's one skipped `)};` sequence at the end of this function right now.